### PR TITLE
More extensive testing, yay!

### PR DESCRIPTION
**Pull request by sunnygleason**
More tests, submitted for #1.

### CODE REVIEW

**Comment by sunnygleason** [comment]
@Licenser I had to add the "simd.rs" file - how do we get around that?

File: src/lib.rs [hunk 1]
```diff
@@ -33,11 +33,12 @@
     external_doc
 )]
 
-#[cfg(target_arch = "aarch64")]
+//#[cfg(target_arch = "aarch64")]
```
Comment (Licenser) [inline src/lib.rs]: we'll need those you can test with: `TARGET=aarch64-unknown-linux-gnu cargo +nightly test` to get it to use the right target

File: src/simd_llvm.rs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 //! TODO: should use `link_llvm_intrinsic` instead: issue #112
 // Taken from stdarch
 
+#[allow(unused)]
```
Comment (Licenser) [inline src/simd_llvm.rs:6]: good call! that was so annoying and I forgot how to get rid of those warnings 😂

**Comment by Licenser** [comment]
For the other tests I got rid of them by using slices that works quite well along with the `arm_cmp` function.

**Comment by Licenser** [comment]
I wonder if we could integrate the test from macros in the specs, that would make creating those complex tests quite easy, any ideas on that or is it too much work?

**Comment by sunnygleason** [comment]
@Licenser I love the idea of updating & improving the tests, I'm definitely up for that. If possible, I'd love to experience the joy of seeing simdjson-rs compile against this library first, then follow up with the testing work?

**Comment by Licenser** [comment]
absolutely! No hurry on that, I was just thinking ahead.

**Pull request merged**